packages/grid/_modules_/grid/hooks/features/columnReorder/useColumnReorder.tsx

```diff
@@ -50,45 +50,60 @@ const hasCursorPositionChanged = (
 ): boolean =>
   currentCoordinates.x !== nextCoordinates.x || currentCoordinates.y !== nextCoordinates.y;
 
-export const useColumnReorder = (columnsRef: React.RefObject<HTMLDivElement>, apiRef: ApiRef) => {
+export const useColumnReorder = (apiRef: ApiRef): void => {
   const logger = useLogger('useColumnReorder');
 
-  const dragCol = React.useRef<ColDef | null>(null);
+  const [, setGridState, forceUpdate] = useGridState(apiRef);
+  const dragCol = useGridSelector(apiRef, columnReorderDragColSelector);
   const dragColNode = React.useRef<HTMLElement | null>(null);
+  const columnsHeaderRef = React.useRef<HTMLElement | null>(null);
   const cursorPosition = React.useRef<CursorCoordinates>({
     x: 0,
     y: 0,
   });
   const removeDnDStylesTimeout = React.useRef<number>();
 
   const handleDragEnd = React.useCallback((): void => {
-    logger.debug(`End dragging col ${dragCol.current!.field}`);
+    logger.debug('End dragging col');
     apiRef.current.publishEvent(COL_REORDER_STOP);
 
     clearTimeout(removeDnDStylesTimeout.current);
 
-    columnsRef.current!.classList.remove(HEADER_CELL_DROP_ZONE_CSS_CLASS);
-    dragColNode.current!.parentElement!.classList.remove('MuiDataGrid-colCellMoving');
+    columnsHeaderRef.current!.classList.remove(HEADER_CELL_DROP_ZONE_CSS_CLASS);
     dragColNode.current!.removeEventListener(DRAGEND, handleDragEnd);
-    dragCol.current = null;
     dragColNode.current = null;
-  }, [columnsRef, apiRef, logger]);
 
-  const handleDragStart = React.useCallback(
+    setGridState((oldState) => {
+      return {
```

Comment:
if you return straightaway no need for the return... 1 row code

Reply:
done